stix_shifter_modules/sentinelone/stix_transmission/ping_connector.py

```diff
@@ -0,0 +1,54 @@
+import json
+import time
+from stix_shifter_utils.modules.base.stix_transmission.base_ping_connector \
+    import BasePingConnector
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from requests.exceptions import ConnectionError
+from stix_shifter_utils.utils import logger
+
+
+class PingConnector(BasePingConnector):
+    """Ping connector class """
+    def __init__(self, api_client):
+        self.api_client = api_client
+        self.logger = logger.set_logger(__name__)
+
+    def ping_connection(self):
+        """
+        Ping the endpoint
+        :return: dict
+        """
+        try:
+            # Construct a response object
+            return_obj = {}
+            response_dict = {}
+
+            return_obj, response_dict = self.call_ping_datasource(return_obj,response_dict)
+        except ConnectionError:
+            response_dict['type'] = "ConnectionError"
+            response_dict['message'] = "Invalid Host"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'])
+        except Exception as ex:
+            if 'Max retries exceeded' in str(ex):
+                # sleep added due to limitation of 1 call a second for each user token
+                time.sleep(1)
+                return_obj, response_dict = self.call_ping_datasource(return_obj, response_dict)
```

Comment:
you also need exception handling after the second ping call

Reply:
Updated the code as suggested.